crates/host/src/module.rs

```diff
@@ -218,13 +279,20 @@ impl SerializedModuleCache {
             })?;
             Ok::<bytes::Bytes, wasmer::RuntimeError>(bytes_mut.into_inner().freeze())
         }) {
-            Some(Ok(serialized_module)) => (
-                unsafe { Module::deserialize(&store, serialized_module.clone()) }
-                    .map_err(|e| wasm_error!(WasmErrorInner::Compile(e.to_string())))?,
-                serialized_module,
-            ),
+            Some(Ok(serialized_module)) => {
+                let deserialized_module =
+                    unsafe { Module::deserialize(&Engine::default(), serialized_module.clone()) }
+                        .map_err(|e| wasm_error!(WasmErrorInner::Compile(e.to_string())))?;
+                (deserialized_module, serialized_module)
+            }
+            // no serialized module found on the file system, so serialize the
+            // wasm binary and write it to the file system
             _fs_miss => {
```

Comment:
Is a file system miss the only option here? What about other errors in a `Some(Err(_))`?

Reply:
I know this is existing code, just checking while we're here and looking at it

Reply:
In case of a Some(Err) the err will be swallowed, something to improve I agree